src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -111,15 +111,16 @@ uct_ib_mlx5_create_cq(uct_ib_iface_t *iface, uct_ib_dir_t dir,
                       const uct_ib_mlx5_iface_config_t *mlx5_config,
                       const uct_ib_iface_config_t *ib_config,
                       const uct_ib_iface_init_attr_t *init_attr,
+                      uct_ib_mlx5_cq_t* mlx5_cq,
                       int preferred_cpu, size_t inl)
 {
+    ucs_status_t status                = UCS_OK;
```

Comment:
Because now status is used in both #if/else branches and even outside of this context.

Reply:
Yes, I understand why you moved it.
The question is, why do we initialize it? It is overridden in both (preprocessor) branches.

Reply:
If we not initialize it, then we need to put it under all other initialized variables. And it would look like:
```c++
#if HAVE_DECL_MLX5DV_CQ_INIT_ATTR_MASK_CQE_SIZE
    uct_ib_device_t *dev               = uct_ib_iface_device(iface);
    struct ibv_cq_init_attr_ex cq_attr = {};
    struct mlx5dv_cq_init_attr dv_attr = {};
    uct_ib_mlx5_md_t *md               = ucs_derived_of(iface->super.md,
                                                        uct_ib_mlx5_md_t);
    ucs_status_t status;
    struct ibv_cq *cq;
    char message[128];
    int cq_errno;
#endif
    ucs_status_t status;
#if HAVE_DECL_MLX5DV_CQ_INIT_ATTR_MASK_CQE_SIZE
...
```

So I preferred just to initialize it to avoid this additional `#if`. Don't you think it is a lesser evil?